tests/services/test_cases.py

```diff
@@ -0,0 +1,119 @@
+        print(mock_connection.post.call_args[0][1])
+        post_data = mock_connection.post.call_args[0][1]
+        assert (
+            post_data["name"] == u"name"
+            and post_data["subject"] == u"subject"
+            and post_data["assignee"] == u"user uid"
+            and post_data["description"] == u"description"
+            and post_data["findings"] == u"findings"
+        )
+
+    def test_get_all_called_with_expected_url_and_default_params(self, mock_connection):
+        cases_service = CasesService(mock_connection)
+        mock_connection.get.side_effect = [
+            DUMMY_GET_ALL_TEST_RESPONSE,
+            DUMMY_GET_ALL_TEST_RESPONSE,
+        ]
+        for _ in cases_service.get_all():
+            continue
+
+        mock_connection.get.call_count == 2
+        assert (
+            mock_connection.get.call_args[0][0]
+            == u"/api/v1/case?pgNum=1&pgSize=500&srtDir=asc&srtKey=number"
+        )
+
+    def test_get_all_called_with_expected_url_and_params(self, mock_connection):
+        cases_service = CasesService(mock_connection)
+        mock_connection.get.side_effect = [
+            DUMMY_GET_ALL_TEST_RESPONSE,
+            DUMMY_GET_ALL_TEST_RESPONSE,
+        ]
+        for _ in cases_service.get_all(name="test-case"):
+            continue
+
+        mock_connection.get.call_count == 2
+        assert (
+            mock_connection.get.call_args[0][0]
+            == u"/api/v1/case?name=test-case&pgNum=1&pgSize=500&srtDir=asc&srtKey=number"
+        )
+
+    def test_get_all_called_with_expected_url_and_all_optional_params(
+        self, mock_connection
+    ):
+        cases_service = CasesService(mock_connection)
+        mock_connection.get.side_effect = [
+            DUMMY_GET_ALL_TEST_RESPONSE,
+            DUMMY_GET_ALL_TEST_RESPONSE,
+        ]
+        for _ in cases_service.get_all(
+            name="test-case",
+            subject="test",
+            assignee="user-uid",
+            updated_at="2010-04-30T001",
+            created_at="2010-01-03T002",
+            status="open",
+        ):
+            continue
+
+        mock_connection.get.call_count == 2
+        assert (
+            mock_connection.get.call_args[0][0]
+            == u"/api/v1/case?name=test-case&subject=test&assignee=user-uid&createdAt=2010-01-03T002&updatedAt=2010-04-30T001&status=open&pgNum=1&pgSize=500&srtDir=asc&srtKey=number"
+        )
+
+    def test_export_called_with_expected_url_and_params(self, mock_connection):
+        cases_service = CasesService(mock_connection)
+        cases_service.export_summary(123456)
+        assert mock_connection.get.call_args[0][0] == u"/api/v1/case/123456/export"
+
+    def test_get_case_by_case_number_called_with_expected_url_and_params(
+        self, mock_connection
+    ):
+        cases_service = CasesService(mock_connection)
+        cases_service.get_case(123456)
+        assert mock_connection.get.call_args[0][0] == u"/api/v1/case/123456"
```

Comment:
These can also use the _TEST_CASE_NUM global param

Reply:
Changed.